### DISCUSSION TITLE
Using @internal/undocumented functions

### DISCUSSION THREAD

**Discussion by hubertlepicki** [InboxSDK/InboxSDK#1109] [Q&A] [answered]
Howdy,

I am workong on a feature that involves adding a custom message to the list of messaages in ThreadView, below the last actual message. I also want to turn Gmail's draft into such message, and remove such draft.

My colleague found two functions which seem to be doing precisely what we need, but they are both undeocumented and 1) is marked as @internal in InboxSDK sources.

The functions I'm talking about are:

1. ThreadView.addCustomMessage()

https://github.com/InboxSDK/InboxSDK/blob/9d7b874e84704f54536cc4dfbbd4281d9e20fa49/src/inboxsdk.d.ts#L414

2. ComposeView.discard()

https://github.com/InboxSDK/InboxSDK/blob/9d7b874e84704f54536cc4dfbbd4281d9e20fa49/src/platform-implementation-js/views/compose-view.ts#L228

So my question is - do you folks know why these functions are not being exposed in docs, are there known issues with them or do you think it's safe to use those?

**Comment by wegry** [comment]
@hubertlepicki hey, thanks for the question. 

1. For `ThreadView.addCustomMessage()`, @Macil or @jfmaggie might be the best reference on whether or not we plan on stabilizing and publicly documenting that method. 

2. For `ComposeView.discard`, we don't have docs, nor do we depend on it directly at Streak. We'd be open to a PR adding it to docs in https://github.com/InboxSDK/inboxsdk-docs.

**Comment by hubertlepicki** [comment]
Okay, thanks @wegry 

About 2) I am happy to do that, but I think I will launch the feature first to my users and see if there are no reports that something doesn't work.

As of 1), I''ll wait for  @Macil or @jfmaggie to chime in

**Comment by jfmaggie** [comment, marked as answer]
Hey @hubertlepicki, thanks for writing in. Sorry for the late response.

for 1), we internally use it ourselves at Streak. So it's safe to use the method if you simply just want to add a custom message but we might not be able to accommodate feature request around the customization right now. We'd welcome you try the method and help improve documentation further given [the source code](https://github.com/InboxSDK/InboxSDK/blob/main/src/platform-implementation-js/dom-driver/gmail/views/gmail-thread-view.ts#L258-L346). Happy to bring it to public with your help here. 

for 2), the compose view discard method itself isn't directly used by Streak but it's used inside InboxSDK mole view management. And you can find usages [here](https://github.com/InboxSDK/InboxSDK/blob/main/src/platform-implementation-js/dom-driver/gmail/widgets/gmail-mole-view-driver.ts#L312) on how we show mole view and [here](https://github.com/InboxSDK/InboxSDK/blob/main/src/platform-implementation-js/dom-driver/gmail/views/gmail-compose-view/override-edit-subject.ts#L62) on how we override editing subject.
